src/core/engine.js

```diff
@@ -16,112 +16,149 @@
+ * @param {object} [config.variables] - An optional list of variables
+ * @param {object} [config.commands] - An optional list of commands
+ * @param {object} [config.plugins] - An optional list of plugins
+ * @param {object} [config.maths] - An optional config for the maths evaluator (@see mathsEvaluator)
  * @returns {calculator}
  */
-function engineFactory({ expression = '', position = 0, maths = {} } = {}) {
+function engineFactory({
+    expression = '',
+    position = null,
+    variables = {},
+    commands = {},
+    plugins = {},
+    maths = {}
+} = {}) {
     /**
-     * Maths expression parser.
-     * @type {function}
+     * The list of event listeners.
+     * @type {Map}
      */
-    let mathsEvaluator = mathsEvaluatorFactory(maths);
+    const events = new Map();
 
     /**
-     * The list of event listeners
+     * A list of variables that can be used in the expression.
      * @type {Map}
      */
-    const events = new Map();
+    const variablesRegistry = new Map();
 
     /**
-     * A list of variables that can be used in the expression
+     * A list of registered commands that can be used inside the calculator.
      * @type {Map}
      */
-    const variables = new Map();
+    const commandsRegistry = new Map();
 
     /**
-     * A list of registered commands that can be used inside the calculator
+     * A list of registered plugins that have been added to the calculator.
+     * It lists the uninstall callback.
      * @type {Map}
      */
-    const commands = new Map();
+    const pluginsRegistry = new Map();
 
     /**
-     * The tokenizer utilized to split down the expression
+     * The tokenizer utilized to split down the expression.
      * @type {calculatorTokenizer}
      */
     const tokenizer = tokenizerFactory();
 
     /**
-     * The list of tokens extracted from the expression
+     * The list of tokens extracted from the expression.
      * @type {Array|null}
      */
     let tokens = null;
 
+    /**
+     * Maths expression parser.
+     * @type {function}
+     */
+    let mathsEvaluator;
+
     /**
      * Engine API
      * @type {object}
      */
     const calculatorApi = {
         /**
          * Registers an event listener.
-         * @param {string} name - The name of the event to listen to.
+         * @param {string} names - The name of the event to listen to. It can be a list separated by spaces.
          * @param {function} listener - The listener to call when the event happen.
          * @returns {calculator} - Chains the instance.
          */
-        on(name, listener) {
-            if ('undefined' !== typeof name && 'function' === typeof listener) {
-                let listeners = events.get(name);
-
-                if (!listeners) {
-                    listeners = new Set();
-                    events.set(name, listeners);
-                }
+        on(names, listener) {
```

Comment:
Not proposing to change it, but is there a reason why the calculator API implements its own `on`/`off`/`trigger` methods instead of using something like `core/eventifier`?

Reply:
Yes, the reason was to reduce the surface of the dependencies. Otherwise, it would require including the whole `tao-core-sdk` inside the LDS for only 3 modules used...